Apps/TimelineApp/feature/timeline/src/main/java/uk/ac/cam/cares/jps/timeline/viewmodel/TrajectoryViewModel.java

```diff
@@ -42,9 +55,31 @@ public void onSuccess(String result) {
 
             @Override
             public void onFailure(Throwable error) {
+                if (error != null) {
+                    LOGGER.error("Trajectory retrieval failed: " + error.getMessage(), error);
+                } else {
+                    LOGGER.error("Trajectory retrieval failed with an unknown error.");
+                }
                 _trajectoryError.postValue(error);
                 _isFetchingTrajectory.postValue(false);
             }
         });
     }
+
+    private long calculateLowerbound(LocalDate date) {
+        ZonedDateTime startOfDay = date.atStartOfDay(ZoneId.of("UTC"));
+        return startOfDay.toInstant().toEpochMilli();
+    }
+
+    private long calculateUpperbound(LocalDate date) {
+        ZonedDateTime endOfDay = date.atTime(23, 59, 59, 999999999)
+                .atZone(ZoneId.of("UTC"));
+        return endOfDay.toInstant().toEpochMilli();
+    }
+
+    private String convertDateFormat(LocalDate date) {
+        ZonedDateTime convertedDateStart = date.atStartOfDay(ZoneId.systemDefault())
+                .withZoneSameInstant(ZoneId.of("UTC"));
+        return convertedDateStart.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSSx"));
```

Comment:
Please use the phone's time zone instead of UTC

Reply:
For now this is fine, in the future you might want to use the time zone of the data, but that might get complex very quickly